### ISSUE TITLE
file encoding issue in Windows related to ui.R and server.R

### ISSUE THREAD

**Issue by shrektan** [rstudio/shiny#1003]
Hi, accoring to [shiny-article-unicode](http://shiny.rstudio.com/articles/unicode.html),

>For the sake of consistency and portability, Shiny requires the character encoding of all its components to be UTF-8, which include ui.R, server.R, global.R, DESCRIPTION, and/or README.md. 

However, in the latest dev version, those files seem to be encoded in native to function well (or the non ASCII character would be shown as a mess). I think it might related to one of your recent commit, since I regularly update my shiny version with the dev version.

#### ui.R
```r
library(shiny)
tagList(
  p("Chinese: 中文")
)
```

#### server.R
```r
function(input, output) {
  
}
```

### output (file encoding **UTF-8**)
![1](https://cloud.githubusercontent.com/assets/8368933/10929603/841ec114-82f0-11e5-9f1a-2523ed99692e.PNG)

### output (file encoding native)
![2](https://cloud.githubusercontent.com/assets/8368933/10929671/4ef699b6-82f1-11e5-8a7b-8e988d397fad.PNG)

### session info
```
> sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: i386-w64-mingw32/i386 (32-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

locale:
[1] LC_COLLATE=Chinese (Simplified)_People's Republic of China.936 
[2] LC_CTYPE=Chinese (Simplified)_People's Republic of China.936   
[3] LC_MONETARY=Chinese (Simplified)_People's Republic of China.936
[4] LC_NUMERIC=C                                                   
[5] LC_TIME=Chinese (Simplified)_People's Republic of China.936    

attached base packages:
[1] stats     graphics  grDevices utils     datasets 
[6] methods   base     

other attached packages:
[1] shiny_0.12.2.9005

loaded via a namespace (and not attached):
 [1] R6_2.1.1         htmltools_0.2.9  tools_3.2.2     
 [4] rstudioapi_0.3.1 Rcpp_0.12.1      jsonlite_0.9.17 
 [7] digest_0.6.8     xtable_1.8-0     httpuv_1.3.3    
[10] mime_0.4 
```

**Comment by yihui** [comment]
Thanks for the report! I can reproduce it, and I'll see if I can fix it tomorrow.

**Comment by shrektan** [comment]
@yihui Thanks. Ha, I'm Xianying Tan, who ask you recommend interns hours ago :dancer:

**Comment by shrektan** [comment]
@yihui Hello, would you please merge your fix into the master branch? The nonsense Chinese characters make me fell annoyed... Thx! :smile:

**Comment by shrektan** [comment]
@yihui your fix 15a49176533045896b249e40ba06fec508fd2d61 will fix the mini example provided above, but when the text contains symbol `\\` it sucks again....

here's the new example.

### UI
```r
library(shiny)
tagList(
  helpText("T:\\2_共享区\\11_资料共享\\MO_Task_File")
)
```

### Server
```r
function(input, output) {
}

```

**Comment by yihui** [comment]
Sorry I have been on sick leave. This issue was trickier than what I imagined. The fix 15a4917 was just a quick experiment, and I knew it was not a complete fix. I'll make another attempt as soon as possible.

**Comment by shrektan** [comment]
Ha, issues related to encoding would never be thin. 

Hope you've fully recovered. Thanks.

**Comment by yihui** [comment]
I have put together a fix in #1025 and you may test it with 

```r
devtools::install_github('yihui/shiny@bugfix/source-utf8')
```

**Comment by shrektan** [comment]
Thanks, it works fine!
